Documentation/data-type-and-flow-control.md

```diff
@@ -0,0 +1,126 @@
+- During transmissions on a stream pipeline, passing through various stream filters, we can guarantee that the same set of input tensors are being processed without the worries of synchronizations after the point of merging or muxing.
+
+The GStreamer pad capability of ```other/tensors``` is as follows:
+```
+other/tensors
+    num_tensors = (int) [1, 16]  # GST_MAX_MEMCHUNK_PER_BUFFER
+    framerate = (fraction) [0/1, 2147483647/1]
+    types = (string) Typestrings
+    dimensions = (string) Dimensions
+
+Typestrings = (string) Typestring
+            | (string) TypeString, TypeStrings
+Typestring = (string) { float32, float64, int64, uint64, int32, uint32, int16, uint16, int8, uint8 }
+Dimensions = (string) Dimension
+           | (string) Dimension, Dimensions
+Dimension = (string) [1-65535]:[1-65535]:[1-65535]:[1-65535]
+```
+
+The buffer of ```other/tensors``` streams have multiple memory chunks. Each memory chunk represents one single tensor in the buffer format of ```other/tensor```. With default configurations of Gstreamer 1.0, the maximum allowed number of memory chunks in a buffer is **16**. Thus, with such configurations of Gstreamer 1.0, ```other/tensors``` may include up to **16** ```other/tensor```.
+
+
+## other/tensorsave (TBU)
+
+```other/tensorsave```, along with its ```typefind``` definition, is defined to enable to save ```other/tensors``` streams as files and load such files are ```other/tensors``` stream. With the definitions of headers defined with ```other/tensorsave```, GStreamer can decode the given file and determine that the file belongs to ```other/tensorsave```.
+
+The detailed description of the file format is at [Design External Save Format for other/tensor and other/tensors Stream for TypeFind](https://github.com/nnstreamer/nnstreamer/wiki/Design-External-Save-Format-for-other-tensor-and-other-tensors-Stream-for-TypeFind)
+
+
+
+# Flow Control
+
+## Timestamps
+
+We have the following principles for timestamp policies. Timestamping policies of ```tensor_*``` filters should follow the given principles.
+
+- Timestamp from the input source (sensors) should be preserved to sink elements.
+- When there are multiple flows merging into one (an element with multiple sink pads), a timestamp of the most recent should be preserved.
+    - For the current frame buffer in a sink pad, ```i``` in ```1 ... n```, ```FB(i)```, and ```T(FB(i))``` is the timestamp of the current frame buffer, the timestamp of the corresponding frame buffer at source pads generated by ```FB(1)``` ... ```FB(n)``` is ```max(i = 1 .. n, T(FB(i)))```, where larger timestamp value means the more recent event.
+    - Example: when multiple frames of the same stream are aggregated by ```tensor_aggregator```, according to this principle, the timestamp of the last frame of the aggregated frames is used for output.

```

Comment:
need update: tensor-aggregator does not have multi pads, and timestamp for outging buffer is not the last frame's.
timestamp of first buffer (+duration)

this example needs update (IMO, tensor-mux is better)

Reply:
Right. It's about `tensor_mux`. 
Modified related descriptions.